crates/red_knot_python_semantic/src/module_resolver/path.rs

```diff
@@ -59,6 +60,27 @@ impl ModulePath {
         self.relative_path.push(component);
     }
 
+    #[must_use]
+    pub(super) fn resolve_pure_module(
+        &self,
+        resolver: &ResolverContext,
+    ) -> Option<(File, ModuleKind)> {
+        let kind = if self.relative_path.ends_with("__init__") {
```

Comment:
Nit: I'm inclined to move this method out of `path` because `path` currently has no *resolution* mehtods. It only has methods to probe the structure. 

I do find the word *pure* unclear in this context. I think I know what you wanted to express with it. Maybe `resolve_file_module`?

Reply:
The intention was to differentiate from `packages` which are also technically modules.
But I also think `resolve_file_module` is better (and achieves the same thing), thanks for the suggestion!

Reply:
Hi @MichaReiser, I've applied your comments:
- I keep a method `is_file_module` in the `ModulePath` to use in the `resolve_package` method
  (I think this fits the pattern of available methods)
- I moved the method `resolve_file_module` to `resolve.rs`

Reply:
Hi @MichaReiser, I've applied your comments:
- I keep a method is_file_module in the ModulePath to use in the resolve_package method
  (I think this fits the pattern of available methods)
- I moved the method resolve_file_module to resolve.rs